packages/authentication/src/metadata.ts

```diff
@@ -0,0 +1,15 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: loopback
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import * as assert from 'assert';
+import 'reflect-metadata';
+
+export function authenticate(controllerClass: any, methodName: string) {
+  Reflect.defineMetadata('loopback:authenticate', true, controllerClass);
```

Comment:
I think we need to make `methodName` optional as follows:
```
methodName?: string
```

I remember that VS code checks the signature of the decorator function to mark if it's valid for class/property/method/parameter.